syntax = "proto3";

package keyvaluestore;

// Interface for key-value store
service KeyValueStore {
  rpc PutBytes (PutBytesRequest) returns (PutBytesReply) {}
  rpc GetBytes (GetBytesRequest) returns (GetBytesReply) {}
}

message PutBytesRequest {
  string key = 1;
  bytes value = 2;
}

message PutBytesReply {
  string key = 1;
}

message GetBytesRequest {
  string key = 1;
}

message GetBytesReply {
  bytes value = 1;
}
